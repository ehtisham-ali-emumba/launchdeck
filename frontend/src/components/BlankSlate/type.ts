export type BlankSlateType = { message?: string };
