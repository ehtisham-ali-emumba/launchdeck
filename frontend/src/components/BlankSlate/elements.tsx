import styled from "styled-components";
import { Typography } from "antd";

const { Text } = Typography;

export const BlankSlateTitle = styled(props => <Text {...props} />)`
  &.ant-typography {
    font-weight: 400;
    color: #797c9a;
    font-size: 18px;
  }
`;
export const BlankSlateWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
  flex-direction: column;
  padding: 0px 16px;
`;
