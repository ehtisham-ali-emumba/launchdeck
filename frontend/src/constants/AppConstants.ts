export const Constants = {
  AUTH_TOKEN_KEY: "authToken",
};
