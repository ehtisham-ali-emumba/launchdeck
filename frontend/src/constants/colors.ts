export const colors = {
  primary: "#ff5c5c",
  accentOrange: "#F16B51",
  neutralGray: "#797c9a",
  chip: "rgb(234, 233, 233)",
  text: {
    primary: "#000000",
    secondary: "#666666",
    light: "#888888",
  },
  black: "#000000",
  background: {
    badge: "#f5f5f5",
  },
  border: "#e8e8e8",
  placeholderColor: "red",
  darkBlue: "#1e2a4a",
  textGray: "#4a4a4a",
  btnSecondary: "#333333",
  white: "#ffffff",
  outline: "#f7774a",
  btnHover: {
    primary: "#e56e50",
    secondary: "#666666",
    outlined: "#fff8f6",
  },
  disabledColor: "#e0e0e0",
};
