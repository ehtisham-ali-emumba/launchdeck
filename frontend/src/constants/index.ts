export * from "./AppConstants";
export * from "./colors";
export * from "./uiStrings";
