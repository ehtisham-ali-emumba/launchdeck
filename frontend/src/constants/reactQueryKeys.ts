export const QUERY_KEYS = {
  TOURS: "tours",
  TOUR_BY_ID: "tourById",
  RANDOM_USERS: "randomUsers",
} as const;
