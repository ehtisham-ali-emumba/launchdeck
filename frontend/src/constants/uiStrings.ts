export const uiStrings = {
  explore: "Explore",
  homepage_title: "The New World With Tourbay",
  homepage_subtitle:
    "No matter where in the world you want to go, we can help get you there and make your tour a stupendous memory.",
  exploreNews: "Explore News",
  tourbay: "Tourbay",
  addTour: "Add Tour",
  bookTour: "Book Tour",
  myTours: "My Tours",
  virtualization: "Virtualization",
  hondaAutos: "Honda Autos",
  brands: "Brands",
  whatsIncluded: "What's Included",
  destination: "Destination",
  departureLoc: "Departure Location",
  return: "Return",
  destinationValue: "Miami",
  departureAddress: "2000 Brush St, Detroit, MI 48226, United States",
  returnTime: "7:00 PM on Day 3",
  additionalInfo:
    "Basic first aid kit Basic first aid kit Basic first aid kit Basic first aid kit",

  bookNow: "Book Now",
  tourDesc: `The Pérez Art Museum Miami (PAMM) is a modern and contemporary art
        museum located in Miami, Florida. It is dedicated to collecting and
        exhibiting international art of the 20th and 21st centuries, with a
        particular focus on works from the Americas, Europe, and Africa.`,
  itinerary: "Itinerary Schedule",
  topDestinations: "Top Destinations",
  deleteConfirmation: "Delete Confirmation",
  deleteConfirmationMessage:
    "Are you sure you want to delete this booking? This action cannot be undone.",
  cancel: "Cancel",
  confirm: "Confirm",
  noToursMessage: "Sorry, We didn’t found any tour right now",
  chooseHere: "Choose Here",
  wantToDeleteBooking: "Are you sure you want to delete this booking?",
  deleteLimitText: ({ title }: { title: string }) =>
    `You can’t delete “${title}” because there are only 3 days remaining untill beginning of this tour. `,
  details: "Details",
  update: "Update",
  days: "days",
  viewDetails: "View Details",
  priceNotAvailable: "Price not available",
  confirmYourBooking: "Confirm Your Booking",
  submitAProduct: "Submit a Product",
  enterYourName: "Enter your name",
  enterYourEmail: "Enter your email",
  enterYourPhone: "Enter your phone",
  numberOfAdults: "Number of Adults",
  numberOfChildren: "Number of Children",
  paymentMethod: "Payment Method",
  creditCard: "Credit Card",
  bankTransfer: "Bank Transfer",
  paypal: "PayPal",
  selectPaymentMethod: "Select payment method",
  popularSearch: "Popular Search",
  priceRange: "Price Range",
  chooseDate: "Choose Date",
  location: "Location",
  whereYouWantToGo: "Where you want to go?",
  searchPlaceholder: "Search by name or email",
  carSearchPlaceholder: "Search by car name or model",
  brandSearchPlaceholder: "Search by brand name",
  by: "By",
  autoNotFound: "Auto not found",
  brandNotExists: "Brand does not exist",
  brandNotFound: "Brand not found",
  carSpecs: "Car Specifications",
  deleteSpecConfirmation: "Would you like to delete this specification?",
  deleteAutoConfirmation: "Would you like to delete this auto?",
  deleteBrandConfirmation: "Would you like to delete this brand?",
  noSpecsFound: "No specifications available",
  noCarsMessage: "No cars available",
  unknownBrand: "Unknown Brand",
  addNewCar: "Add new car",
  enterTheCarName: "Please enter the car name",
  enterTheCarDescription: "Please enter the car description",
  enterCarName: "Enter car name",
  enterCarDescription: "Enter car description",
  plzEnterThePrice: "Please enter the price",
  priceMustBePositive: "Price must be positive",
  enterPrice: "Enter price",
  carName: "Car Name",
  description: "Description",
  price: "Price",
  plzEnterTheBrandName: "Please enter the brand name",
  brandName: "Brand Name",
  enterBrandName: "Enter brand name",
  plzEnterTheDescription: " Please enter the description",
  enterDescription: " Enter description",
  addNewBrand: "Add new brand",
  tourCardImg: "TourCard Image",
  nA: "N/A",
  email: "Email",
  phone: "Phone",
  age: "Age",
  street: "Street",
  city: "City",
  country: "Country",
  close: "Close",

  // launchdeck
  products: "Products",
  landscapes: "Landscapes",
  analytics: "Analytics",
  noDataFound: "Sorry, We didn’t found any data right now",
  submit: "Submit",
  submitProduct: "Submit Product",
  enterName: "Enter product name",
  enterSlug: "Enter product slug",
  productDesc: "Enter product description",
  productWebsite: "Enter product website",
  website: "Website",
  category: "Category",
  categories: "Categories",
  selectProductCategory: "Select Product Category",
  selectProductTags: "Select Product Tags",
  tags: "Tags",
  allLandscapesLoaded: (num: number) => `All ${num} landscapes loaded`,
  allProductsLoaded: (num: number) =>
    `All ${num} product${num > 1 ? "s" : ""} loaded`,
  failedToLoadCategories: "Failed to load categories",
  theBestToTryIn2024: "The best to try in 2024",
  cofounderAt: "Cofounder at",
  askAI:
    "Ask AI about your data... (e.g., 'show me popular health products', 'recent launches')",
  totalProducts: "Total Products",
  avgVotes: "Avg Votes",
  totalVotes: "Total Votes",
  latestLaunch: "Latest Launch",
  analyze: "Analyze",
  clear: "Clear",
  tryThese: "Try these:",
  aiPoweredAnalytics: "AI-Powered Analytics",
  analyticsDiscoverInsights:
    "Discover insights about your product data using natural language queries",
  aiAnalyzingData: "AI is analyzing your data...",
  aiInsights: "AI Insights",
  visualAnalytics: "📊 Visual Analytics",
  dataTables: "📋 Data Tables",
  filteredProducts: (count: number) =>
    `🎯 Filtered Products (${count} results)`,
  allMatchingProducts: "All Matching Products",
  welcomeToAIAnalytics: "Welcome to AI Analytics! 👋",
  startBySearching:
    "Start by searching above to explore your product data with AI-powered insights.",
  askAIAnalyticsPlaceholder:
    "Ask AI about your data... (e.g., 'show me popular products', 'ai tools for product analysis')",
  noDataAvailable: "No data available",
  chartTypeNotSupported: "Chart type not supported",
};
