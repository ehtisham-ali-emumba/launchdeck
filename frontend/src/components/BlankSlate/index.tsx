import { PlaceholderSvg } from "../../assets";
import { uiStrings } from "../../constants";
import { BlankSlateTitle, BlankSlateWrapper } from "./elements";
import type { BlankSlateType } from "./type";

export const BlankSlate: React.FC<BlankSlateType> = ({
  message = uiStrings.noDataFound,
}) => {
  return (
    <BlankSlateWrapper>
      <PlaceholderSvg />
      <BlankSlateTitle>{message}</BlankSlateTitle>
    </BlankSlateWrapper>
  );
};
