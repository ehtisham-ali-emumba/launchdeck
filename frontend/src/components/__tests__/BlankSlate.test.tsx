import { render, screen } from "@testing-library/react";

import { uiStrings } from "../../constants";
import { BlankSlate } from "../BlankSlate";

describe("BlankSlate", () => {
  it("renders with the default message", () => {
    render(<BlankSlate />);
    expect(screen.getByText(uiStrings.noDataFound)).toBeInTheDocument();
    expect(screen.getByTestId("placeholder-svg")).toBeInTheDocument();
  });

  it("renders with a custom message", () => {
    render(<BlankSlate message="Custom message" />);
    expect(screen.getByText("Custom message")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<BlankSlate />);
    console.log("ğŸš€ ~ it ~ container:", container);
    expect(container).toMatchSnapshot();
  });
});
