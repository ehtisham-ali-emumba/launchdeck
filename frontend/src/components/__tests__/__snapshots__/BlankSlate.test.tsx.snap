// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlankSlate > matches snapshot 1`] = `
<div>
  <div
    class="sc-fYsHOw fKBUlR"
  >
    <svg
      data-testid="placeholder-svg"
    />
    <span
      class="ant-typography sc-Qotzb hDzozf css-dev-only-do-not-override-1m63z2v"
    >
      Sorry, We didnâ€™t found any data right now
    </span>
  </div>
</div>
`;
